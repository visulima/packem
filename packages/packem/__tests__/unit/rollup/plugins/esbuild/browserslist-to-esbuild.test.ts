import browserslist from "browserslist";
import { describe, expect, it } from "vitest";

import browserslistToEsbuild from "../../../../../src/rollup/plugins/esbuild/browserslist-to-esbuild";

describe("browserslist-to-esbuild", () => {
    it("works by passing browsers as array", () => {
        expect.assertions(1);

        const target = browserslistToEsbuild(browserslist([">0.2%", "not dead", "not op_mini all"]));

        expect(target).toStrictEqual(["chrome109", "edge133", "firefox115", "ios15.6", "opera116", "safari17.6"]);
    });

    it("works by passing browsers as string", () => {
        expect.assertions(1);

        const target = browserslistToEsbuild(browserslist("last 2 versions"));

        expect(target).toStrictEqual(["chrome135", "edge134", "firefox137", "ie10", "ios18.3", "opera116", "safari18.3"]);
    });

    it("should work with ios", () => {
        expect.assertions(1);

        const target = browserslistToEsbuild(browserslist("ios >= 9"));

        expect(target).toStrictEqual(["ios9"]);
    });

    it("should work with android and ios", () => {
        expect.assertions(1);

        const target = browserslistToEsbuild(browserslist("ios >= 11, android >= 5"));

        expect(target).toStrictEqual(["chrome135", "ios11"]);
    });

    it("should no support android 4", () => {
        expect.assertions(1);

        const target = browserslistToEsbuild(browserslist("android >= 4"));

        expect(target).toStrictEqual(["chrome135"]);
    });
});
