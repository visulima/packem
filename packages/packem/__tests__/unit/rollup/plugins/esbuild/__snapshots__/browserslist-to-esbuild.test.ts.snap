// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`browserslist-to-esbuild > should no support android 4 1`] = `
[
  "chrome138",
]
`;

exports[`browserslist-to-esbuild > should work with android and ios 1`] = `
[
  "chrome138",
  "ios11",
]
`;

exports[`browserslist-to-esbuild > should work with ios 1`] = `
[
  "ios9",
]
`;

exports[`browserslist-to-esbuild > works by passing browsers as array 1`] = `
[
  "chrome109",
  "edge137",
  "firefox140",
  "ios11",
  "safari18.5",
]
`;

exports[`browserslist-to-esbuild > works by passing browsers as string 1`] = `
[
  "chrome138",
  "edge137",
  "firefox140",
  "ie10",
  "ios18.4",
  "opera116",
  "safari18.4",
]
`;
